src/qml/filters/timer/ui.qml

```diff
@@ -178,7 +181,7 @@ Item {
             spacing: 0
             ClockSpinner {
                 id: startSpinner
-                maximumValue: 24 * 60 * 60 // 24 hours
+                maximumValue: 365 * 24 * 60 * 60 // 365 days
```

Comment:
Sure! Looks fine on my machine at least. Typing in `99999999` results in this:

![2021-05-01-16:55:54_3840x2088](https://user-images.githubusercontent.com/1367758/116786330-4f7bcc00-aa9e-11eb-8fa5-e06f99bebc30.png)

The `+` button correctly has no effect in this state, while the `-` button reduces the value to `8759:59:59.983`.

I briefly looked into adding a `DD:...` format too. It looks like components like `ClockSpinner.secondsToTime()` are hard-coded to the assumption that timestamps have exactly three segments, so it won't be easy to support an optional day segment in these input fields. _But_ it seems easy enough to support it as an output format - so for example the Duration spinner might read `72:00:00.000` but the rendered timer in the video can read `01:13:37:00` (1 day, 13 hours, 37 minutes). That looks like it needs a corresponding change in mlt, but it's a small one.

Should the new format(s) pad the days to 2 digits? I'm thinking there should probably be at least one format with no leading zeroes for the days, but I'm not sure how many different pad-widths should be offered.

Reply:
I think it would be easy to add the day indicator to ClockSpinner since you just have to add one extra field in timeToSeconds() and secondsToTime(). But I wonder if it will be obvious what the first field is for. Maybe users will assume the first field is hours and they will have to use trial-and-error to get their desired effect. We could help this by modifying the tooltip.

I agree it is easy to add another format to the output. But now I wonder what format people would want. Would people like to see "05:10:23:51"? I do not think that is really a standard format. Maybe they would rather see "5 Days 10:23:51" or something like that. Also, I kind of think that if you are showing days, you probably don't care about seconds (or maybe even minutes).

How are you using hours > 24 for your project? What format do you use? Are you happy with using large hour numbers?

Reply:
Yes, a mandatory extra field would be easy enough. I was thinking about an optional or context-specific extra field, which would be a bit more complicated, and I don't know how compatible that would be with the rest of the program. I agree that most users probably wouldn't expect there to be a days field, and I think most users wouldn't mind hour counts greater than 24 in the spinner fields.

In my project I'm adding an in-game timer to a sped-up replay of a video game. The timer ends at 59:56:29 (HH:MM:SS), and for this use case I'm perfectly happy with large hour numbers even for the rendered output - I even prefer that over a DD:HH:MM[:SS] format.

Reply:
That is helpful context. Why did you choose 265 days and not 3 days for your use case? Without any need for additional formats, I would support a patch that maxes out the display at 99 hours (99:59:59.999, actually). Would that still meet your needs?

Reply:
Honestly I think it seems a bit arbitrary to have a max value at all (other than the int32 limit at 2^31 frames), so I just picked some "good enough" high value. I don't really see much reason to limit it to <100 hours, but sure, I can make that change if you want.

Reply:
Thanks for your collaboration on this. The reason for the limit is so that we don't have to think through and accommodate all the edge cases and formatting problems that users will inevitably complain about.

Since you are volunteering to do the work, you may choose the resolution:
* Limit the time to 99:59:59
* Add "Day" format for the UI and output (the day field would not have to be optional, but should probably not just be another ":" field since that is not standard)

Reply:
I've pushed the change now, which now makes the spinner fields max out at `99:59:59.999`. Note that the timer output can still go above that, though: for example if you set Offset to `20:00:00` and Duration to `99:59:59.999`, then the timer output will max out at `119:59:59.999`.